src/components/AutoUpdateTime.js

```diff
@@ -22,85 +27,70 @@ const propTypes = {
     ...withLocalizePropTypes,
 };
 
-class AutoUpdateTime extends PureComponent {
-    constructor(props) {
-        super(props);
-        this.getCurrentUserLocalTime = this.getCurrentUserLocalTime.bind(this);
-        this.updateCurrentTime = this.updateCurrentTime.bind(this);
-        this.getTimezoneName = this.getTimezoneName.bind(this);
-        this.state = {
-            currentUserLocalTime: this.getCurrentUserLocalTime(),
-        };
-    }
-
-    componentDidMount() {
-        this.updateCurrentTime();
-    }
-
-    componentDidUpdate() {
-        // Make sure the interval is up to date every time the component updates
-        this.updateCurrentTime();
-    }
-
-    componentWillUnmount() {
-        clearTimeout(this.timer);
-    }
-
+function AutoUpdateTime(props) {
     /**
      * @returns {moment} Returns the locale moment object
      */
-    getCurrentUserLocalTime() {
-        return DateUtils.getLocalMomentFromDatetime(
-            this.props.preferredLocale,
+    const getCurrentUserLocalTime = useCallback(() => (
+        DateUtils.getLocalMomentFromDatetime(
+            props.preferredLocale,
             null,
-            this.props.timezone.selected,
-        );
-    }
+            props.timezone.selected,
+        )
+    ), [props.preferredLocale, props.timezone.selected]);
+
+    const [currentUserLocalTime, setCurrentUserLocalTime] = useState(getCurrentUserLocalTime());
+    const timerRef = useRef(null);
 
     /**
      * @returns {string} Returns the timezone name in string, e.g.: GMT +07
      */
-    getTimezoneName() {
+    const getTimezoneName = useCallback(() => {
         // With non-GMT timezone, moment.zoneAbbr() will return the name of that timezone, so we can use it directly.
-        if (Number.isNaN(Number(this.state.currentUserLocalTime.zoneAbbr()))) {
-            return this.state.currentUserLocalTime.zoneAbbr();
+        if (Number.isNaN(Number(currentUserLocalTime.zoneAbbr()))) {
+            return currentUserLocalTime.zoneAbbr();
         }
 
         // With GMT timezone, moment.zoneAbbr() will return a number, so we need to display it as GMT {abbreviations} format, e.g.: GMT +07
-        return `GMT ${this.state.currentUserLocalTime.zoneAbbr()}`;
-    }
+        return `GMT ${currentUserLocalTime.zoneAbbr()}`;
+    }, [currentUserLocalTime]);
 
     /**
      * Update the user's local time at the top of every minute
      */
-    updateCurrentTime() {
-        if (this.timer) {
-            clearTimeout(this.timer);
-            this.timer = null;
+    const updateCurrentTime = useCallback(() => {
```

Comment:
Since we only reference this inside the effect it's probably better to just pop it straight into the `useEffect()` call.

Reply:
Ah good point. It was a separate function because it was being used in `componentDidMount` and `componentDidUpdate` but we don't have those anymore.